src/modules/ekf2/EKF/ekf_helper.cpp

```diff
@@ -301,131 +301,224 @@ void Ekf::resetAccelBias()
 	resetAccelBiasCov();
 }
 
-void Ekf::get_innovation_test_status(uint16_t &status, float &mag, float &vel, float &pos, float &hgt, float &tas,
-				     float &hagl, float &beta) const
+float Ekf::getHeadingInnovationTestRatio() const
 {
-	// return the integer bitmask containing the consistency check pass/fail status
-	status = _innov_check_fail_status.value;
-
-	// return the largest magnetometer innovation test ratio
-	mag = 0.f;
+	// return the largest heading innovation test ratio
+	float test_ratio = -1.f;
 
 #if defined(CONFIG_EKF2_MAGNETOMETER)
 	if (_control_status.flags.mag_hdg ||_control_status.flags.mag_3D) {
-		mag = math::max(mag, sqrtf(Vector3f(_aid_src_mag.test_ratio).max()));
+		for (auto &test_ratio_filtered : _aid_src_mag.test_ratio_filtered) {
+			test_ratio = math::max(test_ratio, fabsf(test_ratio_filtered));
+		}
 	}
 #endif // CONFIG_EKF2_MAGNETOMETER
 
 #if defined(CONFIG_EKF2_GNSS_YAW)
 	if (_control_status.flags.gps_yaw) {
-		mag = math::max(mag, sqrtf(_aid_src_gnss_yaw.test_ratio));
+		test_ratio = math::max(test_ratio, fabsf(_aid_src_gnss_yaw.test_ratio_filtered));
 	}
 #endif // CONFIG_EKF2_GNSS_YAW
 
 #if defined(CONFIG_EKF2_EXTERNAL_VISION)
 	if (_control_status.flags.ev_yaw) {
-		mag = math::max(mag, sqrtf(_aid_src_ev_yaw.test_ratio));
+		test_ratio = math::max(test_ratio, fabsf(_aid_src_ev_yaw.test_ratio_filtered));
 	}
 #endif // CONFIG_EKF2_EXTERNAL_VISION
 
-	// return the largest velocity and position innovation test ratio
-	vel = NAN;
-	pos = NAN;
+	if (test_ratio >= 0.f && test_ratio <= 1.f) {
+		return sqrtf(test_ratio);
+	}
+
+	return INFINITY;
```

Comment:
Why do you want to return infinity if the test ratio is above 1? If the ratio is valid or not will be decided with a gate at a later point.

Reply:
Right, that doesn't make sense, I think I was being paranoid in a previous iteration with the test ratios initialized to NAN.